### PR TITLE
Add protobuf samples

### PR DESCRIPTION
**Pull request by gierlachg**
Added protobuf samples.

Checklist
- [x] Tags Set
- [x] Milestone Set
- [x] For code samples, code sample main readme is updated

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/2727/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/2727/jet-oss-pr-builder-2727.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:checkstyle (default) on project hazelcast-jet-examples-protobuf: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There are 2 errors reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder_2/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder_2/examples/protobuf/src/main/java/com/hazelcast/jet/examples/protobuf/ProtobufSerializerAdapter.java:1: Line does not match expected header line of '/*'. [Header]
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder_2/examples/protobuf/src/main/java/com/hazelcast/jet/examples/protobuf/ProtobufSerializerHookAdapter.java:1: Line does not match expected header line of '/*'. [Header]
--------------------------
</details>
</pre>

**Comment by cangencer** [comment]
There is an examples readme that needs updating (this was a checklist item..)

**Comment by gierlachg** [comment]
:man_facepalming: Created new section as could not really fit it in into any of existing ones.

File: examples/README.md [hunk 1]
```diff
@@ -210,3 +210,7 @@ samples using the Core API.
   - XML-based dependency injection into a Jet Processor
 - [Spring Boot App that Runs a Jet
   Job](spring-boot/src/main/java/com/hazelcast/jet/examples/spring/SpringBootSample.java)
+
+### Google Proto Buffers
```
Comment (cangencer) [inline examples/README.md:214]: "Protocol Buffers"

**Pull request merged**